packages/fs.go

```diff
@@ -28,32 +28,47 @@ type PackageFileSystem interface {
 // ExtractedPackageFileSystem provides utils to access files in an extracted package.
 type ExtractedPackageFileSystem struct {
 	path string
+	root fs.FS
 }
 
 func NewExtractedPackageFileSystem(p *Package) (*ExtractedPackageFileSystem, error) {
-	return &ExtractedPackageFileSystem{path: p.BasePath}, nil
+	return &ExtractedPackageFileSystem{path: p.BasePath, root: os.DirFS(p.BasePath)}, nil
 }
 
-func (fs *ExtractedPackageFileSystem) Stat(name string) (os.FileInfo, error) {
-	return os.Stat(filepath.Join(fs.path, name))
+func (fsys *ExtractedPackageFileSystem) Stat(name string) (os.FileInfo, error) {
+	name = strings.TrimPrefix(name, "/")
+	return fs.Stat(fsys.root, name)
 }
 
-func (fs *ExtractedPackageFileSystem) Open(name string) (PackageFile, error) {
-	return os.Open(filepath.Join(fs.path, name))
-}
-
-func (fs *ExtractedPackageFileSystem) Glob(pattern string) (matches []string, err error) {
-	matches, err = filepath.Glob(filepath.Join(fs.path, pattern))
+func (fsys *ExtractedPackageFileSystem) Open(name string) (PackageFile, error) {
+	name = strings.TrimPrefix(name, "/")
+	f, err := fsys.root.Open(name)
 	if err != nil {
-		return
+		return nil, err
 	}
-	for i := range matches {
-		match, err := filepath.Rel(fs.path, matches[i])
+	pf, ok := f.(PackageFile)
+	if !ok {
+		defer f.Close()
+		return nil, fmt.Errorf("file does not implement PackageFile interface: %q", name)
```

Comment:
This is a place where we may want to use `fsys.path` to display the full path to the user. Something like this:
```suggestion
		defer f.Close()
		return nil, fmt.Errorf("file does not implement PackageFile interface: %q", filepath.Join(f.path, filepath.FromSlash(name)))
```

If we do this on multiple places, you may consider to implement a helper method for that, something like this:
```
func (fsys *ExtractedPackageFileSystem) fullPath(name string) (string) {
	return filepath.Join(f.path, filepath.FromSlash(name))
}
```